lib/activerecord-import/import.rb

```diff
@@ -704,7 +704,7 @@ def import_helper( *args )
               array_of_attributes[i] = nil
               failure = model.dup
               failure.errors.send(:initialize_dup, model.errors)
-              failed_instances << failure
+              failed_instances << (options[:track_validation_failures] ? [i, failure] : failure )
```

Comment:
Are there scenarios where `i` doesn't match the index of the record in the `failed_instances` array?

Reply:
Apologies for the slow response. As `i` is generated by `arr.each_with_index do |hsh, i|` it should match the index of the record in the original dataset but it will not match the failure's position in `failed_instances`, except for the sad case where all the records have failed.

`arr` maintains the sequence of `array_of_attributes` and, as far as I can tell, `array_of_attributes` maintains the sequence of the original inputs.

Reply:
Ah yes, ok that makes sense.